File: server/express/src/model/taskStatus.ts
import CodeList from "./codeList";

export type TaskStatusCode = string;

export type TaskStatusName = string;

/**
 * Task status code list
 */
export default class TaskStatus extends CodeList<
  TaskStatusCode,
  TaskStatusName
> {
  static get open(): TaskStatusCode {
    return "O";
  }

  static get done(): TaskStatusCode {
    return "D";
  }

  static get cancelled(): TaskStatusCode {
    return "X";
  }
}

File: app/ios/TODOer/Models/Task/TaskStatus.swift
import Foundation

/**
 Status of a `Task` entity
 */
class TaskStatus: CodeList, Comparable {
    
    typealias Code = String
    
    let code: Code
    let name: String
    
    init(code: Code, name: String) {
        self.code = code
        self.name = name
    }
    
    static func == (lhs: TaskStatus, rhs: TaskStatus) -> Bool {
        return lhs.code == rhs.code
    }
    
    static func < (lhs: TaskStatus, rhs: TaskStatus) -> Bool {
        return lhs.code < rhs.code
    }
}
